tests/components/manual_mqtt/test_alarm_control_panel.py

```diff
@@ -11,1772 +10,1698 @@
+        {
+            "alarm_control_panel": {
+                "platform": "manual_mqtt",
+                "name": "test",
+                "code": CODE,
+                "pending_time": 1,
+                "disarm_after_trigger": False,
+                "command_topic": "alarm/command",
+                "state_topic": "alarm/state",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    entity_id = "alarm_control_panel.test"
+
+    assert STATE_ALARM_DISARMED == hass.states.get(entity_id).state
+
+    await common.async_alarm_arm_away(hass, CODE)
+    await hass.async_block_till_done()
+
+    assert STATE_ALARM_PENDING == hass.states.get(entity_id).state
+
+    state = hass.states.get(entity_id)
+    assert state.attributes["post_pending_state"] == STATE_ALARM_ARMED_AWAY
+
+    future = dt_util.utcnow() + timedelta(seconds=1)
+    with patch(
+        ("homeassistant.components.manual_mqtt.alarm_control_panel." "dt_util.utcnow"),
+        return_value=future,
+    ):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+    assert STATE_ALARM_ARMED_AWAY == hass.states.get(entity_id).state
+
+
+async def test_arm_away_with_invalid_code(hass, mqtt_mock):
+    """Attempt to arm away without a valid code."""
+    assert await async_setup_component(
+        hass,
+        alarm_control_panel.DOMAIN,
+        {
+            "alarm_control_panel": {
+                "platform": "manual_mqtt",
+                "name": "test",
+                "code": CODE,
+                "pending_time": 1,
+                "disarm_after_trigger": False,
+                "command_topic": "alarm/command",
+                "state_topic": "alarm/state",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    entity_id = "alarm_control_panel.test"
+
+    assert STATE_ALARM_DISARMED == hass.states.get(entity_id).state
+
+    await common.async_alarm_arm_away(hass, f"{CODE}2")
+    await hass.async_block_till_done()
+
+    assert STATE_ALARM_DISARMED == hass.states.get(entity_id).state
+
+
+async def test_arm_night_no_pending(hass, mqtt_mock):
+    """Test arm night method."""
+    assert await async_setup_component(
+        hass,
+        alarm_control_panel.DOMAIN,
+        {
+            "alarm_control_panel": {
+                "platform": "manual_mqtt",
+                "name": "test",
+                "code": CODE,
+                "pending_time": 0,
+                "disarm_after_trigger": False,
+                "command_topic": "alarm/command",
+                "state_topic": "alarm/state",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    entity_id = "alarm_control_panel.test"
+
+    assert STATE_ALARM_DISARMED == hass.states.get(entity_id).state
+
+    await common.async_alarm_arm_night(hass, CODE, entity_id)
+    await hass.async_block_till_done()
+
+    assert STATE_ALARM_ARMED_NIGHT == hass.states.get(entity_id).state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)